cli/azd/pkg/containerapps/container_app.go

```diff
@@ -256,6 +285,155 @@ func (cas *containerAppService) AddRevision(
 	return nil
 }
 
+func (cas *containerAppService) waitForRevisionReady(
+	ctx context.Context,
+	subscriptionId string,
+	resourceGroupName string,
+	appName string,
+	revisionName string,
+	logProgress func(string)) error {
+	statusPrefix := "Waiting for revision to be ready"
+	logProgress(statusPrefix)
+
+	credential, err := cas.credentialProvider.CredentialForSubscription(ctx, subscriptionId)
+	if err != nil {
+		return err
+	}
+
+	// Add a policy to capture the raw response. This allows us to get fields (ex: 'properties.runningStateDetails') that are
+	// currently not available in armappcontainers.ContainerAppsRevisionsClientGetRevisionResponse
+	responseCapture := azsdk.ResponseCapturePolicy{}
+
+	revisionsClientOptions := *cas.armClientOptions
+	revisionsClientOptions.PerCallPolicies = append(revisionsClientOptions.PerCallPolicies, &responseCapture)
+	revisionsClient, err := armappcontainers.NewContainerAppsRevisionsClient(
+		subscriptionId, credential, &revisionsClientOptions)
+	if err != nil {
+		return fmt.Errorf("creating revisions client: %w", err)
+	}
+
+	replicasClient, err := armappcontainers.NewContainerAppsRevisionReplicasClient(
+		subscriptionId, credential, cas.armClientOptions)
+	if err != nil {
+		return fmt.Errorf("creating replicas client: %w", err)
+	}
+
+	// Poll for the revision to be in a running state
+	// See https://learn.microsoft.com/en-us/azure/container-apps/revisions#lifecycle
+	// for the complete lifecycle of a revision
+	prevStatus := statusPrefix
+	delay := 3 * time.Second
+	pollCount := 0
+	for {
+		revision, err := revisionsClient.GetRevision(ctx, resourceGroupName, appName, revisionName, nil)
+		if err != nil {
+			return fmt.Errorf("getting revision '%s': %w", revisionName, err)
+		}
+
+		if !*revision.Properties.Active {
+			return fmt.Errorf("revision '%s' is not active", revisionName)
+		}
+
+		switch *revision.Properties.RunningState {
+		// Failed states
+		case armappcontainers.RevisionRunningStateFailed,
+			armappcontainers.RevisionRunningStateStopped,
+			armappcontainers.RevisionRunningStateDegraded:
+			errSuffix := ""
+
+			responseExtended := containerAppGetRevisionResponseExtended{}
+			if err := runtime.UnmarshalAsJSON(responseCapture.Response, &responseExtended); err == nil {
+				if responseExtended.Properties.RunningStateDetails != "" {
+					errSuffix = fmt.Sprintf(", %s", responseExtended.Properties.RunningStateDetails)
+				}
+			}
+
+			revisionManagementUrl := output.WithLinkFormat(
+				"%s/#@/resource/subscriptions/%s/resourceGroups/%s/providers/Microsoft.App/containerApps/%s/revisionManagement",
+				string(cas.portalUrlBase),
+				subscriptionId,
+				resourceGroupName,
+				appName)
+
+			if v, err := strconv.ParseBool(os.Getenv("AZD_DEMO_MODE")); err == nil && v {
+				revisionManagementUrl = fmt.Sprintf("Revision Management for %s in Azure Portal", appName)
+			}
```

Comment:
oh, I missed this `AZD_DEMO_MODE` strategy.  I whish demo-mode was not checked everywhere but only within the output (like the console).  IMHO, making all components be aware of azd-demo-mode is not cool... seems unrelated to what component's responsibility is. :(

Reply:
> IMHO, making all components be aware of azd-demo-mode is not cool... seems unrelated to what component's responsibility is. :(

Not opposed to fixing this up over time - putting the checks the few places we needed them vs trying to come up with some generalized console filtering strategy (where perhaps we have a special type for SubscriptionID's and some general thing can see them and filter them) is reasonable for us to go do - I just did the direct thing at the time because there were only a handful of places to audit and it felt like each place needed some level of care to decide "how" to elide the thing.

Perhaps to start on a design there we could just lean into the "I'll give you two options, and demo mode and non demo mode version, and the console decides which to write".

Reply:
I agree that over time we should do something different, but I chose the path of "letting the requirements play out in code over time". This avoids bloat and also really forces us to think about the right way to implement the feature. Like @ellismg, I also think having the flexibility to provide the right substitution at the right place that doesn't degrade UX is a really nice thing to have.

One could also hope for macOS iTerm to provide text hyperlink support, and the rest would be history...

Reply:
> "letting the requirements play out in code over time"

Is that another way to name `tech debt` ? =P

Reply:
In some ways, yes. But I'm not completely sure about the overall impact here... 

Sometimes, waiting for the right moment to implement the right thing with known requirements pays out in long run. But I'll follow up with you to see if anything immediate warrants addressing.